app/server/src/main/scala/org/bitcoins/server/WalletRoutes.scala

```diff
@@ -454,6 +458,56 @@ case class WalletRoutes(wallet: AnyHDWalletApi)(implicit
             val path = walletConf.seedPath
             WalletStorage.changeAesPassword(path, None, Some(password))
 
+            Server.httpSuccess(ujson.Null)
+          }
+      }
+
+    case ServerCommand("importseed", arr) =>
+      ImportSeed.fromJsArr(arr) match {
+        case Failure(err) =>
+          reject(ValidationRejection("failure", Some(err)))
+        case Success(ImportSeed(walletName, mnemonic, passwordOpt)) =>
+          complete {
+            val seedPath = kmConf.seedFolder.resolve(
```

Comment:
can some of this core logic be moved into the key manager project? It seems out of place being in here

Reply:
Not really, it's just making a the seed path and the putting the seed/xprv into our data types